galasa-extensions-parent/dev.galasa.ras.couchdb/src/main/java/dev/galasa/ras/couchdb/internal/CouchdbValidatorImpl.java

```diff
@@ -317,31 +319,31 @@ private void checkVersion(String version, int minVersion, int minRelease, int mi
             actualRelease = Integer.parseInt(m.group(2));
             actualModification = Integer.parseInt(m.group(3));
         } catch (NumberFormatException e) {
-            throw new CouchdbRasException("Unable to determine CouchDB version " + version, e);
+            throw new CouchdbException("Unable to determine CouchDB version " + version, e);
         }
 
         if (actualVersion > minVersion) {
             return;
         }
 
         if (actualVersion < minVersion) {
```

Comment:
I think this method suffers because it's not doing these two steps separately:
- Parsing the string into a CouchDBVersion bean which holds the different parts of a version number.
- The parsing code could go into that class also, maybe on a constructor.
- Comparing the bean with a minimum version